config/v1alpha1/types_insights.go

```diff
@@ -57,15 +55,68 @@ type GatherConfig struct {
+}
+
+// storageSpec provides persistent storage configuration options for on-demand gathering jobs.
+type StorageSpec struct {
+	// type is a required field that specifies the type of storage that will be used to store the Insights data archive.
+	// Valid values are "PersistentVolumeClaim" and "Ephemeral". If the value is omitted, the default value is "Ephemeral".
+	// +optional
+	Type StorageType `json:"type,omitempty"`
+	// persistentVolume is an optional field that specifies the PersistentVolume that will be used to store the Insights data archive.
+	// The PersistentVolume must be created in the openshift-insights namespace.
+	// +optional
+	PersistentVolume PersistentVolumeConfig `json:"persistentVolume,omitempty"`
+}
+
+// storageType declares valid storage types
+// +kubebuilder:validation:Enum=PersistentVolumeClaim;Ephemeral
+type StorageType string
+
+// persistentVolumeConfig provides configuration options for PersistentVolume storage.
+type PersistentVolumeConfig struct {
+	// persistentVolumeClaim is an optional field that specifies the configuration of the PersistentVolumeClaim that will
+	// be used to store the Insights data archive. The PersistentVolumeClaim must be created in the openshift-insights namespace.
+	// +optional
+	PersistentVolumeClaim PersistentVolumeClaimReference `json:"persistentVolumeClaim,omitempty"`
+}
+
+// persistentVolumeClaimReference is a reference to a PersistentVolumeClaim.
+type PersistentVolumeClaimReference struct {
+	// name is a string that follows the DNS1123 subdomain format.
+	// It must be at most 253 characters in length, and must consist only of lower case alphanumeric characters,
+	//  '-' and '.', and must start and end with an alphanumeric character.
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +kubebuilder:validation:MaxLength:=253
+	// +required
+	Name string `json:"name"`
+	// mountPath is an optional field specifying the directory where the PVC will be mounted inside the
+	// Insights data gathering Pod. If omitted, the path that is used to store the Insights data archive by Insights
+	// operator will be used instead. The path cannot exceed 1024 characters and must not contain a colon.
+	// +kubebuilder:validation:MaxLength=1024
+	// +kubebuilder:validation:XValidation:rule="!self.contains(':')",message="mountPath must not contain a colon"
+	// +optional
+	MountPath string `json:"mountPath,omitempty"`
 }
 
 const (
 	// No data obfuscation
 	NoPolicy DataPolicy = "None"
 	// IP addresses and cluster domain name are obfuscated
 	ObfuscateNetworking DataPolicy = "ObfuscateNetworking"
+	// Ephemeral storage type
+	Ephemeral StorageType = "Ephemeral"
+	// PersistentVolumeClaim storage type
+	PersistentVolumeClaim StorageType = "PersistentVolumeClaim"
```

Comment:
It is generally recommended that constants mapping to an enum type alias follow the pattern `{TypeAlias}{Name}`. In this case:
```suggestion
	// StorageTypeEphemeral storage type
	StorageTypeEphemeral StorageType = "Ephemeral"
	// StorageTypePersistentVolumeClaim storage type
	StorageTypePersistentVolumeClaim StorageType = "PersistentVolumeClaim"
```